pandas/_libs/tslibs/parsing.pyx

```diff
@@ -282,7 +282,12 @@ def parse_datetime_string(
         pass
 
     try:
-        dt = du_parse(date_string, default=_DEFAULT_DATETIME,
+        if date_string == "today":
+            dt = datetime.today()
+        elif date_string == "now":
+            dt = datetime.now()
+        else:
+            dt = du_parse(date_string, default=_DEFAULT_DATETIME,
```

Comment:
yes there could be an api change in terms of exception raised.

a] if api change allowed - would this just need some docs change + some exception handling added here ? 
b] or we need to try to maintain compatibility with existing api possible exceptions ?
to ensure same error message class is maintained - should i attempt a try catch in new code block area to ensure we raise the same supported exception type (`TypeError`) ?
again apologies if it's a trivial question, am still bit new here

thanks in advance

Reply:
No problem, the exception class raised by a particular code path should not change unless it's an intentional change. Otherwise it could be considered breaking the API (e.g. a user has pandas code catching a specific exception).

For this block in particular it's not clear to me why we need to add different paths and that might cause different exceptions to be raised downstream?

Reply:
thanks @mroeschke for the reply.
So the above change is to accept "now" and "today" values which are accepted in other parts of parsing ( eg - _parse_today_now in pandas/_libs/tslib.pyx ). Have made a change now to not disturb the core function `parse_datetime_string` but in the relevant invoker of this function in `_array_to_datetime_object` ( in flow 
 of `pd.to_datetime(["today", "yesterday"])` )

Will this change work/be safer ? or do we handle this differently ? apologies if query is something trivial.

Reply:
I would recommend addressing this in a separate PR. It is getting a little difficult tracking all the separate changes being made in this PR.